# new-cap-error

> [ESLint rule][eslint-rules] enforcing that error constructors are invoked with the `new` keyword.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/new-cap-error' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that error constructors are invoked with the `new` keyword.

**Bad**:

<!-- eslint-disable stdlib/new-cap-error -->

```javascript
var err = Error( 'error message' );
```

**Good**:

``` javascript 
var err = new Error( 'error message' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/new-cap-error' );

var linter = new Linter();
var result;

var code = 'var err = Error( \'invoked without new\' );';

linter.defineRule( 'new-cap-error', rule );

result = linter.verify( code, {
    'rules': {
        'new-cap-error': 'error'
    }
});
/* returns
    [
        {
            ruleId: 'new-cap-error',
            severity: 2,
            message: '`Error` constructor is invoked without the `new` keyword',
            line: 1,
            column: 15,
            nodeType: 'CallExpression',
            source: 'var err = Error( \'invoked without new\' );',
            endLine: 1,
            endColumn: 76
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
