/*
* @license Apache-2.0
*
* Copyright (c) 2019 The Stdlib Authors.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import RandomStream = require( './index' );


// TESTS //

// The constructor returns a readable stream...
{
	// tslint:disable: no-unused-expression
	new RandomStream( 2.0, 5.0 ); // $ExpectType RandomStream
	new RandomStream( 2.0, 5.0, {} ); // $ExpectType RandomStream

	// tslint:enable: no-unused-expression
}

// The compiler throws an error if the constructor is provided invalid input arguments...
{
	const s1 = new RandomStream( '2.0', 5.0 ); // $ExpectError
}

// The compiler throws an error if the constructor is provided insufficient arguments...
{
	const s1 = new RandomStream(); // $ExpectError
	const s2 = new RandomStream( 2.0 ); // $ExpectError
}
