# Blockquote Indentation

> [ESLint rule][eslint-rules] to enforce Markdown blockquote indentation in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-blockquote-indentation' );
```

#### rule

[ESLint rule][eslint-rules] to enforce Markdown blockquote indentation in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-blockquote-indentation, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep.
*
* >   This is a blockquote.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep.
*
* > This is a blockquote.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

The [rule][eslint-rules] may be configured using the same options as supported by [remark][remark-lint-blockquote-indentation]. By default, the rules enforces an indentation of `2` characters, which includes the `>` symbol.

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-blockquote-indentation' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '* >    This is an indented blockquote.',
    '*',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-blockquote-indentation', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-blockquote-indentation': [ 'error', 2 ]
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-blockquote-indentation',
            'severity': 2,
            'message': 'Remove 3 spaces between blockquote and content',
            'line': 4,
            'column': 6,
            'nodeType': null,
            'source': '/**',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

[remark-lint-blockquote-indentation]: https://github.com/remarkjs/remark-lint/tree/19150d94f89f7a0d94d083417890236d11839641/packages/remark-lint-blockquote-indentation

</section>

<!-- /.links -->
