/*
* @license Apache-2.0
*
* Copyright (c) 2021 The Stdlib Authors.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import arc4 = require( './index' );


// TESTS //

// The function returns a cipher...
{
	arc4( 'pwd12' ); // $ExpectType Cipher
}

// The compiler throws an error if the function is provided an argument other than a string...
{
	arc4( 123 ); // $ExpectError
	arc4( true ); // $ExpectError
	arc4( false ); // $ExpectError
	arc4( [] ); // $ExpectError
	arc4( {} ); // $ExpectError
	arc4( ( x: number ): number => x ); // $ExpectError
}

// The compiler throws an error if the function is provided an unsupported number of arguments...
{
	arc4(); // $ExpectError
	arc4( 'abc', {} ); // $ExpectError
}

// The compiler throws an error if the returned cipher is provided an invalid argument...
{
	const fcn = arc4( 'beep' );
	fcn( 123 ); // $ExpectError
	fcn( true ); // $ExpectError
	fcn( false ); // $ExpectError
	fcn( [] ); // $ExpectError
	fcn( {} ); // $ExpectError
	fcn( ( x: number ): number => x ); // $ExpectError
}

// The compiler throws an error if the returned cipher is provided an unsupported number of arguments...
{
	const fcn = arc4( 'beep' );
	fcn(); // $ExpectError
	fcn( 'abc', {} ); // $ExpectError
}
