<!--

@license Apache-2.0

Copyright (c) 2018 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# module-exports-assignments

> [ESLint rule][eslint-rules] enforcing that only variables and functions are assigned to `module.exports`.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/module-exports-assignments' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that only variables and functions are assigned to `module.exports`.

**Bad**:

<!-- eslint-disable stdlib/module-exports-assignments -->

```javascript
var normal = require( '@stdlib/random/base/normal' );
var beta = require( '@stdlib/random/base/beta' );


// EXPORTS //

module.exports = {
    'normal': normal,
    'beta': beta
};
```

**Good**:

```javascript
var normal = require( '@stdlib/random/base/normal' );
var beta = require( '@stdlib/random/base/beta' );


// VARIABLES //

var ns = {
    'normal': normal,
    'beta': beta
};


// EXPORTS //

module.exports = ns;
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/module-exports-assignments' );

var linter = new Linter();
var result;
var code;

code = [
    'module.exports = fun();',
    '',
    'function fun() {',
    '  return 3.14;',
    '}'
].join( '\n' );

linter.defineRule( 'module-exports-assignments', rule );

result = linter.verify( code, {
    'rules': {
        'module-exports-assignments': 'error'
    }
});
/* returns
    [
        {
            'ruleId': 'module-exports-assignments',
            'severity': 2,
            'message': 'Only variables and functions should be assigned to `module.exports`',
            'line': 1,
            'column': 1,
            'nodeType': 'AssignmentExpression',
            'source': 'module.exports = fun();',
            'endLine': 1,
            'endColumn': 23
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
