# require-order

> [ESLint rule][eslint-rules] enforcing a specified order of `require()` calls.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/require-order' );
```

#### rule

[ESLint rule][eslint-rules] enforcing a specified order of `require()` calls. By default, require statements should appear in the following order:

-   **builtin**: built-in Node.js modules.

-   **external**: external packages.

-   **path**: relative or absolute paths (strings beginning with `./`, `/`, or `../`.
)

To change the default tag ordering, set the `order` option of the rule. Besides `builtin`, `external`, and `path`, the `order` array can hold regular expression strings such as `/^@stdlib/`.

**Bad**:

<!-- eslint-disable stdlib/require-order -->

```javascript
/*
    Options: [ 'error', {
        'order': [ 'builtin', 'external', '/^@stdlib/', 'path' ]
    }]
*/
var validate = require( './validate.js' );
var math = require( '@stdlib/math' );
var debug = require( 'debug' );
var fs = require( 'fs' );
```

**Good**:

``` javascript
/*
    Options: [ 'error', {
        'order': [ 'builtin', 'external', '/^@stdlib/', 'path' ]
    }]
*/
var fs = require( 'fs' );
var debug = require( 'debug' );
var math = require( '@stdlib/math' );
var validate = require( './validate.js' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/require-order' );

var linter = new Linter();
var result;

var code = [
    'var validate = require( \'./validate.js\' )',
    'var fs = require( \'fs\' )'
].join( '\n' );

linter.defineRule( 'require-order', rule );

result = linter.verify( code, {
    'rules': {
        'require-order': 'error'
    }
});
/* returns
    [
        {
            'ruleId': 'require-order',
            'severity': 2,
            'message': '`require( \'fs\' )` should come before `require( \'./validate.js\' )`',
            'line': 2,
            'column': 10,
            'nodeType': null,
            'source': 'var fs = require( \'fs\' )',
            'endLine': 2,
            'endColumn': 25
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
