# section-headers

> [ESLint rule][eslint-rules] to enforce formatting of section header comments.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/section-headers' );
```

#### rule

[ESLint rule][eslint-rules] to enforce formatting of section header comments. Section header comments must be in uppercase letters and end with `//`. The following headers are allowed:

-   MODULES
-   VARIABLES
-   FUNCTIONS
-   MAIN
-   EXPORTS
-   NOTES
-   FIXTURES
-   TESTS

**Bad**:

<!-- eslint-disable stdlib/section-headers -->

```javascript
// FUNCTIOns //

function add( x, y ) {
    return x + y;
}


// CONSTANTS //

var TWO_PI = 2.0 * Math.PI;


// EXPORTS

module.exports = {};
```

**Good**:

```javascript
// VARIABLES //

var TWO_PI = 2.0 * Math.PI;


// EXPORTS //

module.exports = TWO_PI;
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/section-headers' );

console.log( rule );
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
