# require-file-extensions

> [ESLint rule][eslint-rules] enforcing that `require()` calls of files end with a whitelisted file extension.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/require-file-extensions' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that `require()` calls of files end with a whitelisted file extension.

**Bad**:

<!-- eslint-disable stdlib/require-file-extensions -->

```javascript
// Invalid file extension:
var readme = require( '@stdlib/types/array/int32/README.md' );

// Missing file extension:
var debug = require( 'debug/src/browser' );
```

**Good**:

``` javascript 
var Int32Array = require( '@stdlib/types/array/int32' );

var debug = require( 'debug/src/browser.js' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/require-file-extensions' );

console.log( rule );
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
