# no-internal-require

> [ESLint rule][eslint-rules] disallowing require() calls into internals of another stdlib package.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-internal-require' );
```

#### rule

[ESLint rule][eslint-rules] disallowing require() calls into internals of another stdlib package.

<!-- eslint-disable -->

```javascript
// Bad...
var betainc = require( '@stdlib/math/base/special/betainc/lib/betainc.js' );

// Good...
var betainc = require( '@stdlib/math/base/special/betainc' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-internal-require' );

console.log( rule );
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
