<!--

@license Apache-2.0

Copyright (c) 2018 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# no-native-math

> [ESLint rule][eslint-rules] enforcing that `stdlib` equivalents are used instead of the global `Math` object. Exceptions are made for `Math` functions that currently do not have a `stdlib` equivalent.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-native-math' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that `stdlib` equivalents are used instead of the global `Math` object.

**Bad**:

<!-- eslint-disable stdlib/no-native-math -->

```javascript
var out = Math.exp( 2.0 );
// returns ~7.389
```

**Good**:

``` javascript 
var exp = require( '@stdlib/math/base/special/exp' );
var out = exp( 2.0 );
// returns ~7.389
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/no-native-math' );

var linter = new Linter();
var result;
var code;

code = 'var y = Math.sqrt( 9.0 );';

linter.defineRule( 'no-native-math', rule );

result = linter.verify( code, {
    'rules': {
        'no-native-math': 'error'
    }
});
/* returns
    [
        {
            'ruleId': 'no-native-math',
            'severity': 2,
            'message': 'Use `var sqrt = require( \'@stdlib/math/base/special/sqrt\' )` instead of the `Math.sqrt` method',
            'line': 1,
            'column': 9,
            'nodeType': 'MemberExpression',
            'source': 'var y = Math.sqrt( 9.0 );',
            'endLine': 1,
            'endColumn': 18
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
