# Table Pipe Alignment

> [ESLint rule][eslint-rules] to require table pipes to be aligned in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-table-pipe-alignment' );
```

#### rule

[ESLint rule][eslint-rules] to require table pipes to be aligned in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-table-pipe-alignment, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Boop beep.
*
* | x | y |
* | any | boop |
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Boop beep.
*
* | x   | y    |
* | any | boop |
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-table-pipe-alignment' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Squares a number.',
    '*',
    '* | Input | Output |',
    '* | ----- | ------ |',
    '* | 2     | 4 |',
    '*',
    '* @param {number} x - input value',
    '* @returns {number} output value',
    '*',
    '* @example',
    '* var out = square( 2 );',
    '* // returns 4',
    '*/',
    'function square( x ) {',
    '\treturn x*x;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-table-pipe-alignment', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-table-pipe-alignment': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            ruleId: 'jsdoc-table-pipe-alignment',
            severity: 2,
            message: 'Misaligned table fence',
            line: 6,
            column: 15,
            nodeType: null,
            source: '* | 2     | 4 |',
            endLine: 14,
            endColumn: 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
