# require-typed-arrays

> [ESLint rule][eslint-rules] enforcing that typed array constructors are explicitly required.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/require-typed-arrays' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that typed array constructors are explicitly required.

**Bad**:

<!-- eslint-disable stdlib/require-typed-arrays -->

```javascript
var arr = new Float32Array();
```

**Good**:

``` javascript 
var Float32Array = require( '@stdlib/types/array/float32' );

var arr = new Float32Array();
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/require-typed-arrays' );

console.log( rule );
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
