# Definition Case

> [ESLint rule][eslint-rules] to require Markdown definition labels in JSDoc descriptions to be lowercased.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-definition-case' );
```

#### rule

[ESLint rule][eslint-rules] to require Markdown definition labels in JSDoc descriptions to be lowercased.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-definition-case, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep [boop][Boop].
*
* [Boop]: http://example.com
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep [boop][boop].
*
* [boop]: http://example.com
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-definition-case' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep [boop][Boop].',
    '*',
    '* [Boop]: http://example.com',
    '*',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-definition-case', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-definition-case': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-definition-case',
            'severity': 2,
            'message': 'Do not use upper-case characters in definition labels',
            'line': 4,
            'column': 3,
            'nodeType': null,
            'source': '* [Boop]: http://example.com',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
