# no-nested-require

> [ESLint rule][eslint-rules] enforcing that one does not use nested property access for `require()` expressions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-nested-require' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that one does not use nested property access for `require()` expressions.

**Bad**:

<!-- eslint-disable stdlib/no-nested-require -->

```javascript
var special = require( '@stdlib' ).math.base.special;
```

**Good**:

``` javascript 
var special = require( '@stdlib/math/base' ).special;

special = require( '@stdlib/math/base/special' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/no-nested-require' );

var linter = new Linter();
var result;
var code;

code = 'var special = require( \'@stdlib/math\' ).base.special;';

linter.defineRule( 'no-nested-require', rule );

result = linter.verify( code, {
    'rules': {
        'no-nested-require': 'error'
    }
});
/* =>
    [
        {
            ruleId: 'no-nested-require',
            severity: 2,
            message: 'do not use nested property access for require() expressions',
            line: 2,
            column: 15,
            nodeType: 'CallExpression',
            source: 'var special = require( \'@stdlib/math\' ).base.special;',
            endLine: 2,
            endColumn: 29
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
