<!--

@license Apache-2.0

Copyright (c) 2018 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# no-require-indexjs

> [ESLint rule][eslint-rules] disallowing `require()` calls to explicitly require `index.js` files.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-require-indexjs' );
```

#### rule

[ESLint rule][eslint-rules] disallowing `require()` calls to explicitly require `index.js` files.

**Bad**:

<!-- eslint-disable stdlib/no-require-indexjs -->

```javascript
var debug = require( 'debug/index.js' );
```

**Good**:

``` javascript 
var debug = require( 'debug' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/no-require-indexjs' );

var linter = new Linter();
var result;

var code = 'var abs = require( \'@stdlib/math/base/special/abs/lib/index.js\' );';

linter.defineRule( 'no-require-indexjs', rule );

result = linter.verify( code, {
    'rules': {
        'no-require-indexjs': 'error'
    }
});
/* returns
    [
        {
            'ruleId': 'no-require-indexjs',
            'severity': 2,
            'message': 'require() call should omit filename from path if equal to `index.js`',
            'line': 1,
            'column': 11,
            'nodeType': 'CallExpression',
            'source': 'var abs = require( \'@stdlib/math/base/special/abs/lib/index.js\' );',
            'endLine': 1,
            'endColumn': 66
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
