# Inline Padding

> [ESLint rule][eslint-rules] to prevent padding of inline nodes between their markers and content in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-inline-padding' );
```

#### rule

[ESLint rule][eslint-rules] to prevent padding of inline nodes between their markers and content in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-no-inline-padding, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Boop: _ beep _.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Boop: _beep_.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-inline-padding' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep: _ boop _.',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-no-inline-padding', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-no-inline-padding': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-no-inline-padding',
            'severity': 2,
            'message': 'Donâ€™t pad `emphasis` with inner spaces',
            'line': 2,
            'column': 9,
            'nodeType': null,
            'source': '* Beep: _ boop _.',
            'endLine': 10,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
