# no-unassigned-require

> [ESLint rule][eslint-rules] forbidding unassigned require statements.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-unassigned-require' );
```

#### rule

[ESLint rule][eslint-rules] forbidding unassigned require statements.

**Bad**:

<!-- eslint-disable stdlib/no-unassigned-require -->

```javascript
require( 'tape' );
```

**Good**:

``` javascript 
var tape = require( 'tape' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/no-unassigned-require' );

var linter = new Linter();
var result;

var code = 'require( \'@stdlib/math\' );';

linter.defineRule( 'no-unassigned-require', rule );

result = linter.verify( code, {
    'rules': {
        'no-unassigned-require': 'error'
    }
});
/* returns
    [
        {
            'ruleId': 'no-unassigned-require',
            'severity': 2,
            'message': 'result of `require()` call is not assigned',
            'line': 1,
            'column': 15,
            'nodeType': 'ExpressionStatement',
            'source': 'require( \'@stdlib/math\' );',
            'endLine': 1,
            'endColumn': 27
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
