# Code Block Style

> [ESLint rule][eslint-rules] to require that Markdown code blocks adhere to a specific style in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-code-block-style' );
```

#### rule

[ESLint rule][eslint-rules] to require that Markdown code blocks adhere to a specific style in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-code-block-style, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep.
*
*     y = x;
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep.
*
* ```javascript
* y = x;
* ```
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

The [rule][eslint-rules] may be configured using the same options as supported by [remark][remark-lint-code-block-style]. By default, the rule requires that code blocks be demarcated using fences.

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-code-block-style' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '*     var x = y;',
    '*',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-code-block-style', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-code-block-style': [ 'error', 'fenced' ]
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-code-block-style',
            'severity': 2,
            'message': 'Code blocks should be fenced',
            'line': 4,
            'column': 3,
            'nodeType': null,
            'source': '*     var x = y;',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

[remark-lint-code-block-style]: https://github.com/remarkjs/remark-lint/tree/19150d94f89f7a0d94d083417890236d11839641/packages/remark-lint-code-block-style

</section>

<!-- /.links -->
