# no-require-absolute-path

> [ESLint rule][eslint-rules] disallowing `require()` calls of absolute file paths.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-require-absolute-path' );
```

#### rule

[ESLint rule][eslint-rules] disallowing `require()` calls of absolute file paths.

**Bad**:

<!-- run-disable -->

<!-- eslint-disable stdlib/no-require-absolute-path -->

```javascript
var stdlib = require( '/path/to/stdlib' );
```

**Good**:

``` javascript 
var stdlib = require( '@stdlib' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/no-require-absolute-path' );

var linter = new Linter();
var result;

var code = 'var stdlib = require( \'/path/to/stdlib/\' );';

linter.defineRule( 'no-require-absolute-path', rule );

result = linter.verify( code, {
    'rules': {
        'no-require-absolute-path': 'error'
    }
});
/* returns
    [
        {
            'ruleId': 'no-require-absolute-path',
            'severity': 2,
            'message': 'require() call contains an absolute file path',
            'line': 1,
            'column': 14,
            'nodeType': 'CallExpression',
            'source': 'var stdlib = require( \'/path/to/stdlib/\' );',
            'endLine': 1,
            'endColumn': 43
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
