# Literal URLs

> [ESLint rule][eslint-rules] to prevent URLs without angle-brackets from being used in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-literal-urls' );
```

#### rule

[ESLint rule][eslint-rules] to prevent URLs without angle-brackets from being used in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-no-literal-urls, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Boop beep.
*
* ## Links
*
* -   http://foo.bar/baz
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Boop beep.
*
* ## Links
*
* -   <http://foo.bar/baz>
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-literal-urls' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '* ## Links',
    '*',
    '* -   http://foo.bar/baz',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-no-literal-urls', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-no-literal-urls': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-no-literal-urls',
            'severity': 2,
            'message': 'Donâ€™t use literal URLs without angle brackets',
            'line': 6,
            'column': 7,
            'nodeType': null,
            'source': '* -   http://foo.bar/baz',
            'endLine': 14,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
