/*
* @license Apache-2.0
*
* Copyright (c) 2020 The Stdlib Authors.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import ddot = require( './index' );


// TESTS //

// The function returns a number...
{
	ddot( {}, {} ); // $ExpectType number
}

// The compiler throws an error if the function is provided a first argument which is not an ndarray...
{
	// FIXME: add ndarray arguments
	// ddot( 10, x ); // $ExpectError
	// ddot( '10', x ); // $ExpectError
	// ddot( true, x ); // $ExpectError
	// ddot( false, x ); // $ExpectError
	// ddot( null, x ); // $ExpectError
	// ddot( undefined, x ); // $ExpectError
	// ddot( [], x ); // $ExpectError
	// ddot( {}, x ); // $ExpectError
	// ddot( ( x: number ): number => x, x ); // $ExpectError
}

// The compiler throws an error if the function is provided a second argument which is not an ndarray...
{
	// FIXME: add ndarray arguments
	// ddot( x, 10 ); // $ExpectError
	// ddot( x, '10' ); // $ExpectError
	// ddot( x, true ); // $ExpectError
	// ddot( x, false ); // $ExpectError
	// ddot( x, null ); // $ExpectError
	// ddot( x, undefined ); // $ExpectError
	// ddot( x, [] ); // $ExpectError
	// ddot( x, {} ); // $ExpectError
	// ddot( x, ( x: number ): number => x ); // $ExpectError
}

// The compiler throws an error if the function is provided an unsupported number of arguments...
{
	// FIXME: add ndarray arguments
	ddot(); // $ExpectError
	ddot( {} ); // $ExpectError
	ddot( {}, {}, 1 ); // $ExpectError
}
