# require-globals

> [ESLint rule][eslint-rules] enforcing that specified global variables are explicitly required.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/require-globals' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that specified global variable are explicitly required. The list of globals has to be

**Bad**:

<!-- eslint-disable stdlib/require-globals -->

```javascript
/*
    Options: [ 'error', {
        'globals': [
            'Float32Array'
        ]
    }]
*/
var arr = new Float32Array();
```

**Good**:

<!-- eslint-disable stdlib/require-globals -->

``` javascript 
/*
    Options: [ 'error', {
        'globals': [
            'Float32Array'
        ]
    }]
*/
var Float32Array = require( '@stdlib/array/float32' );

var arr = new Float32Array();

arr = new Int32Array();
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/require-globals' );

var linter = new Linter();
var result;

var code = 'var arr = new Int32Array( [1,2,3] );';

linter.defineRule( 'require-globals', rule );

result = linter.verify( code, {
    'rules': {
        'require-globals': [ 'error', {
            'globals': [
                'Int32Array'
            ]
        }]
    }
});
/* returns
    [
        {
            ruleId: 'require-globals',
            severity: 2,
            message: 'Int32Array is invoked without loading it beforehand via require()',
            line: 1,
            column: 11,
            nodeType: 'NewExpression',
            source: 'var arr = new Int32Array( [1,2,3] );',
            endLine: 1,
            endColumn: 36
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
