# Heading-Like Paragraph

> [ESLint rule][eslint-rules] to prevent paragraphs which appear to be h7+ "headings" in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-heading-like-paragraph' );
```

#### rule

[ESLint rule][eslint-rules] to prevent paragraphs which appear to be h7+ "headings" in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-no-heading-like-paragraph, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Boop beep.
*
* ####### Boop
*
* Beep.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Boop beep.
*
* ## Boop
*
* Beep.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-heading-like-paragraph' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '* ####### Beep',
    '*',
    '* boop',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-no-heading-like-paragraph', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-no-heading-like-paragraph': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-no-heading-like-paragraph',
            'severity': 2,
            'message': 'This looks like a heading but has too many hashes',
            'line': 4,
            'column': 3,
            'nodeType': null,
            'source': '* ####### Beep',
            'endLine': 14,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
