/*
* @license Apache-2.0
*
* Copyright (c) 2022 The Stdlib Authors.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import get = require( './index' );


// TESTS //

// The function returns a value...
{
	get( [ 1, 2, 3, 4 ], 2 ); // $ExpectType any
}

// The compiler throws an error if the function is provided a first argument which is not a collection...
{
	get( 5, 2 ); // $ExpectError
	get( true, 2 ); // $ExpectError
	get( false, 2 ); // $ExpectError
	get( null, 2 ); // $ExpectError
	get( {}, 2 ); // $ExpectError
}

// The compiler throws an error if the function is provided a second argument which is not a number...
{
	get( [], '5' ); // $ExpectError
	get( [], true ); // $ExpectError
	get( [], false ); // $ExpectError
	get( [], null ); // $ExpectError
	get( [], {} ); // $ExpectError
	get( [], [] ); // $ExpectError
	get( [], ( x: number ): number => x ); // $ExpectError
}

// The compiler throws an error if the function is provided an unsupported number of arguments...
{
	get(); // $ExpectError
	get( [] ); // $ExpectError
	get( [], 1, 2 ); // $ExpectError
}
