# jsdoc-leading-description-sentence

> [ESLint rule][eslint-rules] to enforce that JSDoc descriptions start with an uppercase letter or number and end with a period.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-leading-description-sentence' );
```

#### rule

[ESLint rule][eslint-rules] to enforce that JSDoc descriptions start with an uppercase letter or number and end with a period. To supply an array of words to be allowed at the beginning of a JSDoc description without being capitalized, set the `whitelist` option of the rule.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-leading-description-sentence -->

```javascript
/**
* returns a pseudo-random number on `[0,1]`
*
* @returns {number} uniform random number
*
* @example
* var y = rand();
* // e.g., returns 0.5363925252089496
*/
function rand() {
    return Math.random();
}
```

**Good**:

```javascript
/**
* Returns a pseudo-random number on `[0,1]`.
*
* @returns {number} uniform random number
*
* @example
* var y = rand();
* // e.g., returns 0.5363925252089496
*/
function rand() {
    return Math.random();
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-leading-description-sentence' );

var linter = new Linter();
var result;
var code;

code = [
    '/**',
    '* squares a number.',
    '* ',
    '* @param {number} x - input number',
    '* @returns {number} x squared',
    '*',
    '* @example',
    '* var y = square( 2.0 );',
    '* // returns 4.0',
    '*/',
    'function square( x ) {',
    '  return x*x;',
    '}'
].join( '\n' );

linter.defineRule( 'jsdoc-leading-description-sentence', rule );

result = linter.verify( code, {
    'rules': {
        'jsdoc-leading-description-sentence': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-leading-description-sentence',
            'severity': 2,
            'message': 'Description must start with an uppercase letter or number and end with a period',
            'line': 1,
            'column': 1,
            'nodeType': null,
            'source': '/**',
            'endLine': 10,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
