# Fenced Code Marker

> [ESLint rule][eslint-rules] to enforce a specific Markdown fenced code marker in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-fenced-code-marker' );
```

#### rule

[ESLint rule][eslint-rules] to enforce a specific Markdown fenced code marker in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-fenced-code-marker, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep boop.
*
* ~~~javascript
* y = x;
* ~~~
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep boop.
*
* ```javascript
* y = x;
* ```
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

The [rule][eslint-rules] may be configured using the same options as supported by [remark][remark-lint-fenced-code-marker]. By default, the rule enforces that `\`` be used as the Markdown fenced code marker in JSDoc descriptions.

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-fenced-code-marker' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep *boop*.',
    '*',
    '* ~~~javascript',
    '* y = x;',
    '* ~~~',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-fenced-code-marker', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-fenced-code-marker': [ 'error', '`' ]
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-fenced-code-marker',
            'severity': 2,
            'message': 'Fenced code should use ` as a marker',
            'line': 4,
            'column': 3,
            'nodeType': null,
            'source': '* ~~~javascript',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

[remark-lint-fenced-code-marker]: https://github.com/remarkjs/remark-lint/tree/19150d94f89f7a0d94d083417890236d11839641/packages/remark-lint-fenced-code-marker

</section>

<!-- /.links -->
