# Unordered List Marker Style

> [ESLint rule][eslint-rules] to enforce a specified Markdown unordered list marker style in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-unordered-list-marker-style' );
```

#### rule

[ESLint rule][eslint-rules] to enforce a specified Markdown unordered list marker style in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-unordered-list-marker-style, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep boop.
*
* +   Foo
* +   Bar
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep boop.
*
* -   Foo
* -   Bar
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

The [rule][eslint-rules] may be configured using the same options as supported by [remark][remark-lint-unordered-list-marker-style]. By default, the rule enforces that `-` be used as the Markdown unordered list marker in JSDoc descriptions.

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-unordered-list-marker-style' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep *boop*.',
    '*',
    '* +   Foo',
    '* +   Bar',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-unordered-list-marker-style', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-unordered-list-marker-style': [ 'error', '-' ]
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-unordered-list-marker-style',
            'severity': 2,
            'message': 'Marker style should be `-`',
            'line': 4,
            'column': 3,
            'nodeType': null,
            'source': '* +   Foo',
            'endLine': 13,
            'endColumn': 3
        },
                {
            'ruleId': 'jsdoc-unordered-list-marker-style',
            'severity': 2,
            'message': 'Marker style should be `-`',
            'line': 5,
            'column': 3,
            'nodeType': null,
            'source': '* +   Bar',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

[remark-lint-unordered-list-marker-style]: https://github.com/remarkjs/remark-lint/tree/19150d94f89f7a0d94d083417890236d11839641/packages/remark-lint-unordered-list-marker-style

</section>

<!-- /.links -->
