# no-redeclare

> [ESLint rule][eslint-rules] disallowing variable redeclarations.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-redeclare' );
```

#### rule

[ESLint rule][eslint-rules] disallowing variable redeclarations.

**Bad**:

<!-- eslint-disable stdlib/no-redeclare -->

```javascript
var a = 'beep';

// ...

var a = 'boop';
```

**Good**:

``` javascript 
var a = 'beep';

// ...

a = 'boop';
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-redeclare' );

console.log( rule );
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
