# no-redeclare

> [ESLint rule][eslint-rules] disallowing variable redeclarations.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-redeclare' );
```

#### rule

[ESLint rule][eslint-rules] disallowing variable redeclarations.

**Bad**:

<!-- eslint-disable stdlib/no-redeclare -->

```javascript
var a = 'beep';

// ...

var a = 'boop';
```

**Good**:

``` javascript 
var a = 'beep';

// ...

a = 'boop';
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/no-redeclare' );

var linter = new Linter();
var result;

var code = 'var Object = 0';

linter.defineRule( 'no-redeclare', rule );

result = linter.verify( code, {
    'rules': {
        'no-redeclare': [ 'error', {
            'builtinGlobals': true
        }]
    }
});
/* returns
    [
        {
            'ruleId': 'no-redeclare',
            'severity': 2,
            'message': 'Object is already defined',
            'line': 1,
            'column': 5,
            'nodeType': 'Identifier',
            'source': 'var Object = 0',
            'endLine': 1,
            'endColumn': 11
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
