# Hard Break Spaces

> [ESLint rule][eslint-rules] to prevent too many spaces from being used to hard break in JSDoc descriptions containing Markdown.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-hard-break-spaces' );
```

#### rule

[ESLint rule][eslint-rules] to prevent too many spaces from being used to hard break in JSDoc descriptions containing Markdown.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-hard-break-spaces, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep boop.
*
* Dots represent⋅⋅⋅
* spaces.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep boop.
*
* Dots represent⋅⋅
* spaces.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-hard-break-spaces' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '* Additional   ',
    '* information.',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-hard-break-spaces', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-hard-break-spaces': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-hard-break-spaces',
            'severity': 2,
            'message': 'Use two spaces for hard line breaks',
            'line': 4,
            'column': 13,
            'nodeType': null,
            'source': '* Additional   ',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
