# No Tabs

> [ESLint rule][eslint-rules] forbidding the use of tabs in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-tabs' );
```

#### rule

[ESLint rule][eslint-rules] forbidding the use of tabs in JSDoc descriptions.

**Bad**:

<!--lint disable no-tabs-->

<!-- eslint-disable stdlib/jsdoc-no-tabs, stdlib/jsdoc-markdown-remark, no-tabs -->

```javascript
/**
* Beep.
*
* -	List item starting with a tab.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep.
*
* -   List item starting with spaces.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

<!-- eslint-disable no-tabs -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-tabs' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '* -	This is a list item starting with a tab.',
    '*',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-no-tabs', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-no-tabs': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-no-tabs',
            'severity': 2,
            'message': 'Use spaces instead of hard-tabs',
            'line': 4,
            'column': 4,
            'nodeType': null,
            'source': '* -\tThis is a list item starting with a tab.',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
