# Table Indentation

> [ESLint rule][eslint-rules] to prevent unneeded indentation before tables in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-table-indentation' );
```

#### rule

[ESLint rule][eslint-rules] to prevent unneeded indentation before tables in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-no-table-indentation, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Boop beep.
*
*   | x   | y    |
*   | any | boop |
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Boop beep.
*
* | x   | y    |
* | any | boop |
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-table-indentation' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Squares a number.',
    '*',
    '* Input | Output',
    '* ----- | ------',
    '* 2     | 4',
    '*',
    '* @param {number} x - input value',
    '* @returns {number} output value',
    '*',
    '* @example',
    '* var out = square( 2 );',
    '* // returns 4',
    '*/',
    'function square( x ) {',
    '\treturn x*x;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-no-table-indentation', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-no-table-indentation': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-no-table-indentation',
            'severity': 2,
            'message': 'Do not indent table rows',
            'line': 4,
            'column': 3,
            'nodeType': null,
            'source': '*    | Input | Output |',
            'endLine': 14,
            'endColumn': 3
        },
        {
            'ruleId': 'jsdoc-no-table-indentation',
            'severity': 2,
            'message: 'Do not indent table rows',
            'line': 6,
            'column': 3,
            'nodeType': null,
            'source': '*    | 2     | 4      |',
            'endLine': 14,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
