# Strong Marker

> [ESLint rule][eslint-rules] to enforce a specific Markdown strong marker in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-strong-marker' );
```

#### rule

[ESLint rule][eslint-rules] to enforce a specific Markdown strong marker in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-strong-marker, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep __boop__.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep **boop**.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

The [rule][eslint-rules] may be configured using the same options as supported by [remark][remark-lint-strong-marker]. By default, the rule enforces that `*` be used as the Markdown strong marker in JSDoc descriptions.

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-strong-marker' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep __boop__.',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-strong-marker', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-strong-marker': [ 'error', '*' ]
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-strong-marker',
            'severity': 2,
            'message': 'Strong should use `*` as a marker',
            'line': 2,
            'column': 8,
            'nodeType': null,
            'source': '* Beep __boop__.',
            'endLine': 10,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

[remark-lint-strong-marker]: https://github.com/remarkjs/remark-lint/tree/19150d94f89f7a0d94d083417890236d11839641/packages/remark-lint-strong-marker

</section>

<!-- /.links -->
