# Ordered List Marker Value

> [ESLint rule][eslint-rules] to enforce that marker values of ordered lists do not violate a given style in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-ordered-list-marker-value' );
```

#### rule

[ESLint rule][eslint-rules] to enforce that marker values of ordered lists do not violate a given style in JSDoc descriptions. By default,list item marker values should be incremented by one.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-ordered-list-marker-value, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep boop.
*
* 1.  Foo
* 1.  Bar
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep boop.
*
* 1.  Foo
* 2.  Bar
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

The [rule][eslint-rules] may be configured using the same options as supported by [remark][remark-lint-ordered-list-marker-value]. Options besides the default, `ordered`, are `single` and `one`. `single` enforces that all bullet values are the same as the starting point, whereas `one` enforces that bullet values are always equal to `1`.

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-ordered-list-marker-value' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep *boop*.',
    '*',
    '* 1.  Foo',
    '* 1.  Bar',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-ordered-list-marker-value', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-ordered-list-marker-value': [ 'error', 'ordered' ]
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-ordered-list-marker-value',
            'severity': 2,
            'message': 'Marker should be `2`, was `1`',
            'line': 5,
            'column': 3,
            'nodeType': null,
            'source': '* 1.  Bar',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

[remark-lint-ordered-list-marker-value]: https://github.com/remarkjs/remark-lint/tree/19150d94f89f7a0d94d083417890236d11839641/packages/remark-lint-ordered-list-marker-value

</section>

<!-- /.links -->
