# new-cap-array

> [ESLint rule][eslint-rules] enforcing that the `Array` constructor is invoked with the `new` keyword.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/new-cap-array' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that the `Array` constructor are invoked with the `new` keyword.

**Bad**:

<!-- eslint-disable stdlib/new-cap-array -->

```javascript
var arr = Array( 101 );
```

**Good**:

``` javascript 
var arr = new Array( 101 );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/new-cap-array' );

var linter = new Linter();
var result;

var code = 'var arr = Array( 10 );';

linter.defineRule( 'new-cap-array', rule );

result = linter.verify( code, {
    'rules': {
        'new-cap-array': 'error'
    }
});
/* returns
    [
        {
            ruleId: 'new-cap-array',
            severity: 2,
            message: '`Array` constructor is invoked without the `new` keyword',
            line: 1,
            column: 11,
            nodeType: 'CallExpression',
            source: 'var arr = Array( 10 );',
            endLine: 1,
            endColumn: 41
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
