# no-dynamic-require

> [ESLint rule][eslint-rules] enforcing that require() calls have only string literals as arguments.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-dynamic-require' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that require() calls have only string literals as arguments.

**Bad**:

<!-- eslint-disable stdlib/no-dynamic-require -->

```javascript
var pkg = '@stdlib/math/base/special/betainc';
var betainc = require( pkg );
```

**Good**:

``` javascript 
var betainc = require( '@stdlib/math/base/special/betainc' );
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/no-dynamic-require' );

console.log( rule );
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
