# List Item Bullet Indent

> [ESLint rule][eslint-rules] to prevent unnecessary indentation of Markdown list item bullets in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-list-item-bullet-indent' );
```

#### rule

[ESLint rule][eslint-rules] to prevent unnecessary indentation of Markdown list item bullets in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-list-item-bullet-indent, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep.
*
*   -   Item
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep.
*
* -   Item
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-list-item-bullet-indent' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '*   -   Item',
    '*',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-list-item-bullet-indent', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-list-item-bullet-indent': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-list-item-bullet-indent',
            'severity': 2,
            'message': 'Incorrect indentation before bullet: remove 2 spaces',
            'line': 4,
            'column': 7,
            'nodeType': null,
            'source': '*   -   Item',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
