# Checkbox Content Indent

> [ESLint rule][eslint-rules] to prevent Markdown checkboxes being followed by too much whitespace in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-checkbox-content-indent' );
```

#### rule

[ESLint rule][eslint-rules] to prevent Markdown checkboxes being followed by too much whitespace in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-checkbox-content-indent, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep.
*
* -   [ ]   Item
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep.
*
* -   [ ] Item
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-checkbox-content-indent' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '* -   [ ]    Item',
    '*',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-checkbox-content-indent', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-checkbox-content-indent': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-checkbox-content-indent',
            'severity': 2,
            'message': 'Checkboxes should be followed by a single character',
            'line': 4,
            'column': 12,
            'nodeType': null,
            'source': '/**',
            'endLine': 13,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
