<!--

@license Apache-2.0

Copyright (c) 2018 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# ternary-conditional-parantheses

> [ESLint rule][eslint-rules] enforcing that boolean expressions are wrapped in parantheses when used as conditions in a ternary operator.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/ternary-conditional-parantheses' );
```

#### rule

[ESLint rule][eslint-rules] enforcing that boolean expressions are wrapped in parantheses when used as conditions in a ternary operator.

**Bad**:

<!-- eslint-disable stdlib/ternary-conditional-parantheses -->

```javascript
var randu = require( '@stdlib/random/base/randu' );

var bool = randu() > 0.5 ? 1 : 0;
```

**Good**:

``` javascript 
var randu = require( '@stdlib/random/base/randu' );

var bool = ( randu() > 0.5 ) ? 1 : 0;
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/ternary-conditional-parantheses' );

var linter = new Linter();
var result;

var code = 'var bool = randu() > 0.5 ? 1 : 0;';

linter.defineRule( 'ternary-conditional-parantheses', rule );

result = linter.verify( code, {
    'rules': {
        'ternary-conditional-parantheses': 'error'
    }
});
/* returns
    [
        {
            'ruleId': 'ternary-conditional-parantheses',
            'severity': 2,
            'message': 'boolean expression should be wrapped in parantheses',
            'line': 1,
            'column': 12,
            'nodeType': 'BinaryExpression',
            'source': 'var bool = randu() > 0.5 ? 1 : 0;',
            'endLine': 1,
            'endColumn': 25
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
