# Blockquote Marker

> [ESLint rule][eslint-rules] to require blank lines in Markdown blockquotes have a marker in JSDoc descriptions.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-blockquote-without-marker' );
```

#### rule

[ESLint rule][eslint-rules] to require blank lines in Markdown blockquotes have a marker in JSDoc descriptions.

**Bad**:

<!-- eslint-disable stdlib/jsdoc-no-blockquote-without-marker, stdlib/jsdoc-markdown-remark -->

```javascript
/**
* Beep.
*
* > This is a...
*
* > ...blockquote.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

**Good**:

```javascript
/**
* Beep.
*
* > This is a...
* >
* > ...blockquote.
*
* @returns {string} a value
*
* @example
* var str = beep();
* // returns 'boop'
*/
function beep() {
    return 'boop';
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/jsdoc-no-blockquote-without-marker' );

var linter = new Linter();
var result;
var code;

// Generate our source code:
code = [
    '/**',
    '* Beep boop.',
    '*',
    '* > Beep...',
    '*',
    '* > ...boop.',
    '*',
    '*',
    '* @param {string} str - input value',
    '* @returns {string} output value',
    '*',
    '* @example',
    '* var out = beep( "boop" );',
    '* // returns "beepboop"',
    '*/',
    'function beep( str ) {',
    '\treturn "beep" + str;',
    '}'
].join( '\n' );

// Register the ESLint rule:
linter.defineRule( 'jsdoc-no-blockquote-without-marker', rule );

// Lint the code:
result = linter.verify( code, {
    'rules': {
        'jsdoc-no-blockquote-without-marker': 'error'
    }
});
console.log( result );
/* =>
    [
        {
            'ruleId': 'jsdoc-no-blockquote-without-marker',
            'severity': 2,
            'message': 'Missing marker in blockquote',
            'line': 5,
            'column': 3,
            'nodeType': null,
            'source': '*',
            'endLine': 15,
            'endColumn': 3
        }
    ]
*/
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
