# empty-line-before-comment

> [ESLint rule][eslint-rules] to enforce empty lines before comments.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/empty-line-before-comment' );
```

#### rule

[ESLint rule][eslint-rules] to enforce empty lines before comments.

**Bad**:

<!-- eslint-disable stdlib/empty-line-before-comment -->

```javascript
function square( x ) {
    var out;
    // Square the number:
    out = x*x;
    return out;
}
```

**Good**:

```javascript
function square( x ) {
    var out;

    // Square the number:
    out = x*x;
    return out;
}
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/empty-line-before-comment' );

console.log( rule );
```

</section>

<!-- /.examples -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
